pkg/plugins/document/dynamodb/dynamodb.go

```diff
@@ -556,20 +580,37 @@ func isBetweenEnd(index int, exps []document.QueryExpression) bool {
 	return false
 }
 
-func getTableName(collection document.Collection) *string {
+func (s *DynamoDocService) getTableName(collection document.Collection) (*string, error) {
 	coll := collection
 	for coll.Parent != nil {
 		coll = *coll.Parent.Collection
 	}
 
-	return aws.String(coll.Name)
+	// TODO: The following method for determining the deployment specific table name from the nitric name is unreliable.
+	//	a new design is in process and will replace this method for locating tables.
+	out, err := s.client.ListTables(&dynamodb.ListTablesInput{})
+
```

Comment:
I think we should be caching these table name look results in memory. We can have a retry if the table value is not found, was not found to handle any deployment timing issues.

Reply:
Good suggestion, I've added some basic caching now